src/enc_effects.c

```diff
@@ -0,0 +1,366 @@
+    {ELITE_FOUR_ENC_1,      SEQ_BA_TENNO},          // Aaron
+    {ELITE_FOUR_ENC_2,      SEQ_BA_TENNO},          // Bertha
+    {ELITE_FOUR_ENC_3,      SEQ_BA_TENNO},          // Flint
+    {ELITE_FOUR_ENC_4,      SEQ_BA_TENNO},          // Lucian
+
+    // Champion
+    {CHAMPION_ENC,          SEQ_BA_CHANP},          // Champion (Cynthia)
+
+    // Rival
+    {CATCHALL_ENC,          SEQ_BA_RIVAL},          // Rival (Barry/Cedric)
+    
+    // Legendary Pokemon
+    {LEGENDARY_POKE_ENC_1,  SEQ_BA_POKE},           // Shaymin
+    {LEGENDARY_POKE_ENC_2,  SEQ_BA_DPOKE2},         // Dialga, Palkia
+    {LEGENDARY_POKE_ENC_2,  SEQ_BA_DPOKE1},         // Uxie, Azelf
+    {CATCHALL_ENC,          SEQ_BA_DPOKE1},         // Mesprit (roaming)
+    {LEGENDARY_POKE_ENC_2,  SEQ_BA_SECRET1},        // Arceus
+    {LEGENDARY_POKE_ENC_1,  SEQ_BA_SECRET2},        // Regigigas, Heatran
+    {CATCHALL_ENC,          SEQ_BA_POKE},           // Cresselia (roaming)
+    {CATCHALL_ENC,          SEQ_BA_SECRET2},        // Articuno, Zapdos, Moltres (roaming)
+    {LEGENDARY_POKE_ENC_1,  SEQ_PL_BA_GIRA},        // Giratina
+    {LEGENDARY_POKE_ENC_1,  SEQ_PL_BA_REGI},        // Regirock, Regice, Registeel
+
+    // Team Galactic
+    {GALACTIC_GRUNT_ENC,    SEQ_BA_GINGA},          // Grunts
+    {GALACTIC_LEADER_ENC,   SEQ_BA_GINGA3},         // Commanders (Mars, Jupiter, Saturn)
+    {GALACTIC_LEADER_ENC,   SEQ_BA_AKAGI},          // Cyrus
+
+    // Other special encounter types
+    {FRONTIER_ENC,          SEQ_BA_TRAIN},          // Frontier trainers
+    {FRONTIER_ENC,          SEQ_BA_TRAIN},          // Link/Wi-Fi battles
+    {DOUBLE_ENC,            SEQ_BA_TRAIN},          // Double-battle against a trainer
+    {DOUBLE_ENC,            SEQ_BA_POKE},           // Wild encounters with a partner NPC
+    {FRONTIER_ENC,          SEQ_PL_BA_BRAIN},       // Frontier Brains (Palmer, Thorton, Argenta, Dahlia, Darach)
+    {DOUBLE_ENC,            SEQ_BA_GYM},            // Double-battle against a gym leader (Volkner+Flint, specifically)
+
+    // All other encounters
+    {CATCHALL_ENC,          SEQ_BA_TRAIN},          // Generic trainer
+    {CATCHALL_ENC,          SEQ_BA_POKE}            // Generic wild Pokemon
+};
+
+static u32 EncEffects_GetEffectsForTrainerClass(u32 trainerClass);
+static u32 EncEffects_GetEffectsForWildMon(Party * wildParty, int zoneID);
+static u32 EncEffects_GetEffectsForBattle(const UnkStruct_ov6_02240D5C * battleData);
+static u32 EncEffects_GetCutInFromEffects(u32 effectsID, const UnkStruct_ov6_02240D5C * battleData);
+static u32 EncEffects_GetBGMFromEffects(u32 effectsID, const UnkStruct_ov6_02240D5C * battleData);
+
+
+// Fight type flag definitions
+// The fight type structure is a sequence of bits that tell us stuff
+// about the encounter
+#define FIGHT_TYPE_POKEMON           0      // bit 1 off -> this is a fight against a Pokemon
+#define FIGHT_TYPE_TRAINER           1      // bit 1 on -> this is a fight against a trainer
```

Comment:
The use of define names here is inconsistent. While the rest of defines are all for individual bits (either set or unset), these two first defines are the possible *values* of the first bit.

There's different ways to go about this:

1. Make them all individual on-off values you can OR together
```
u32 fight_type = FIGHT_TYPE_TRAINER | FIGHT_TYPE_DOUBLE_OFF | FIGHT_TYPE_DOUBLE_OFF | FIGHT_TYPE_FRONTIER_ON;
```

2. Make them all bit flags, disable them by simply not specifying them
```
u32 fight_type = FIGHT_TYPE_TRAINER_F | FIGHT_TYPE_FRONTIER_F;
```

3. Keep the current special treatment of the first bit as a *value* define rather than a flag define, but make it clear in the naming:
```
u32 fight_type = FIGHT_TYPE_TRAINER_VAL | FIGHT_TYPE_FRONTIER_F;
```
(the `_VAL` part is optional, added for clarity)

The choice here primarily depends on how special the first bit is.